statsmodels/genmod/tests/test_glm.py

```diff
@@ -1021,6 +1022,200 @@ def test_gradient_irls():
                    assert_allclose(gradient_bse, rslt_irls.bse, rtol=1e-6, atol=1e-6)
 
 
+class CheckWtdDuplicationMixin(object):
+    decimal_params = DECIMAL_4
+
+    def test_params(self):
+        assert_allclose(self.res1.params, self.res2.params, 1e-6)
+
+    decimal_bse = DECIMAL_4
+
+    def test_standard_errors(self):
+        assert_allclose(self.res1.bse, self.res2.bse, 1e-6)
+
+    decimal_resids = DECIMAL_4
+
+    # TODO: This doesn't work... Arrays are of different shape.
+    # Perhaps we use self.res1.model.family.resid_XXX()?
+    """
+    def test_residuals(self):
+        resids1 = np.column_stack((self.res1.resid_pearson,
+                                   self.res1.resid_deviance,
+                                   self.res1.resid_working,
+                                   self.res1.resid_anscombe,
+                                   self.res1.resid_response))
+        resids2 = np.column_stack((self.res1.resid_pearson,
+                                   self.res2.resid_deviance,
+                                   self.res2.resid_working,
+                                   self.res2.resid_anscombe,
+                                   self.res2.resid_response))
+        assert_allclose(resids1, resids2, self.decimal_resids)
+    """
+
+    def test_aic(self):
+        # R includes the estimation of the scale as a lost dof
+        # Doesn't with Gamma though
+        assert_allclose(self.res1.aic, self.res2.aic, 1e-6)
+
+    def test_deviance(self):
+        assert_allclose(self.res1.deviance, self.res2.deviance, 1e-6)
+
+    def test_scale(self):
+        assert_allclose(self.res1.scale, self.res2.scale, 1e-6)
+
+    def test_loglike(self):
+        # Stata uses the below llf for these families
+        # We differ with R for them
+        assert_allclose(self.res1.llf, self.res2.llf, 1e-6)
+
+    decimal_null_deviance = DECIMAL_4
+
+    def test_null_deviance(self):
+        assert_allclose(self.res1.null_deviance, self.res2.null_deviance, 1e-6)
+
+    decimal_bic = DECIMAL_4
+
+    def test_bic(self):
+        assert_allclose(self.res1.bic, self.res2.bic, 1e-6)
+
+    decimal_fittedvalues = DECIMAL_4
+
+    def test_fittedvalues(self):
+        res2_fitted = self.res2.predict(self.res1.model.exog)
+        assert_allclose(self.res1.fittedvalues, res2_fitted, 1e-6)
+
+    decimal_tpvalues = DECIMAL_4
+
+    def test_tpvalues(self):
+        # test comparing tvalues and pvalues with normal implementation
+        # make sure they use normal distribution (inherited in results class)
+        assert_allclose(self.res1.tvalues, self.res2.tvalues, 1e-6)
+        assert_allclose(self.res1.pvalues, self.res2.pvalues, 1e-6)
```

Comment:
This needs an atol because some values are very small. (problem with using only relative tolerance if the value is essentially zero)
And in general it's better to use the keyword `rtol` instead of relying on positional interpretation

`assert_allclose(self.res1.pvalues, self.res2.pvalues, rtol=1e-6, atol=1e-20)`

i.e. for example atol=1e-20: if the pvalue is smaller than 1e-20, then we don't care how close to zero.